src/react-components/layout/SpinWhileTrue.js

```diff
@@ -0,0 +1,22 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Spinner } from "../misc/Spinner";
+import { Center } from "./Center";
+
+export function SpinWhileTrue({ isSpinning, children }) {
```

Comment:
Honestly the ternary was what I had before, but with a bunch of the components using it the same way, I thought pulling it out into a component was more readable. The ternary made it look messy especially since most of the components has <>{error && <p></p>}<I am other components>.... </>.